packages/redeem_solver/src/service.ts

```diff
@@ -0,0 +1,97 @@
+import type { Utxo, RedeemRequest } from "@gonative-cc/lib/types";
+import type { Storage } from "./storage";
+import type { SuiClient } from "./sui_client";
+import { logger, logError } from "@gonative-cc/lib/logger";
+import type { SuiNet } from "@gonative-cc/lib/nsui";
+
+export class RedeemService {
+	constructor(
+		private storage: Storage,
+		private clients: Map<SuiNet, SuiClient>,
+	) {}
+
+	async processPendingRedeems() {
+		const pendingRequests = await this.storage.getPendingRedeems();
+		if (pendingRequests.length === 0) {
+			logger.info({ msg: "No pending redeem requests" });
+			return;
+		}
+
+		for (const req of pendingRequests) {
+			await this.processRequest(req);
+		}
+	}
+
+	private async processRequest(req: RedeemRequest) {
+		logger.info({
+			msg: "Processing redeem request",
+			redeemId: req.redeem_id,
+			amountSats: req.amount_sats.toString(),
+		});
+		const availableUtxos = await this.storage.getAvailableUtxos(req.package_id);
+		const selectedUtxos = this.selectUtxos(availableUtxos, req.amount_sats);
+
+		if (!selectedUtxos) {
+			logger.warn({
+				msg: "Insufficient UTXOs for request",
+				redeemId: req.redeem_id,
+				neededAmountSats: req.amount_sats.toString(),
+			});
+			return;
+		}
+
+		const client = this.clients.get(req.sui_network);
```

Comment:
make a helper function as we have in btcindexer, with proper validation for pkg ids in the constructor

https://github.com/gonative-cc/workers/blob/master/packages/btcindexer/src/btcindexer.ts#L129